Some Misc. changes

Rajat163 opened the pull request.

Rajat163 committed:
Some Misc. changes

drishti07 commented on src/main/java/com/rajat/mywebapp/controllers/CustomerController.java:
```diff
@@ -32,7 +32,7 @@ public List<CustomerDTO> getAllCustomers() {
     }
 
-    @RequestMapping(params = "custId",method = RequestMethod.GET)
+    @GetMapping("customers/id")
```
There is a mismatch in the parameter mentioned in the get mapping and in the function argument.

drishti07 replied:
Also if id is variable like from front end it should be like customers/24 then it should be like following:
@GetMapping("/{id}")

drishti07 commented on src/main/java/com/rajat/mywebapp/service/impl/CustomerServiceImpl.java:
```diff
@@ -23,14 +23,14 @@ public class CustomerServiceImpl implements CustomerService {
 
     @Override
-    public void addNewCustomer(CustomerDTO dto) {
+    public long addNewCustomer(CustomerDTO dto) {
```
suggestion: put the code in try and catch blocks and handle exceptions.

drishti07 commented on src/main/resources/application.properties:
```diff
@@ -2,6 +2,6 @@ spring.datasource.url=jdbc:mysql://localhost:3306/customer_manager
 spring.datasource.password=S@xena1216
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
-spring.jpa.hibernate.ddl-auto=update
+spring.jpa.hibernate.ddl-auto=create
```
I hope this is due to the fact that if you do any changes in db then it is dropped and created again?

Rajat163 replied:
Reverting this one for higher env.

Rajat163 committed:
1. Updating the endpoints as per rest best practices
2. Adding the exception handling for service layer
3. Update customer functionality not working
4. Changed properties file for local and higher envs
